src/ResourceStore.php

```diff
@@ -107,17 +122,29 @@ public function find($type, $id, array $query_params = [])
     // }}}
     // {{{ public function query()
 
+    /**
+     * Finds a resource from the API server.
+     *
+     * @param string $type         the resource type to find.
+     * @param string $id           the resource identifier.
+     * @param array  $query_params optional array of extra parameters.
+     *
+     * @return Resource|null the resource object or null if no such resource
+     *         could be found.
+     */
     public function query($type, $id, array $query_params = [])
     {
-        $result = $this->http_client->request(
-            'GET',
-            $this->getResourceAddress($type, $id),
-            ['query' => $query_params]
-        );
-
-        $body = json_decode($result->getBody(), true);
+        $resource = null;
 
-        $this->checkJsonBody($body);
+        try {
+            $body = $this->doRequest(
+                'GET',
+                $this->getResourceAddress($type, $id),
+                ['query' => $query_params]
+            );
+        } catch (ResourceNotFoundException $e) {
+            // not found is non-fatal for query method
```

Comment:
We should be returning `null` here. Right now we continue on to the trying to build the resource.